pkg/command/restore/dcmappings.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2025 ScyllaDB
+
+package restore
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type dcMappings []dcMapping
+
+type dcMapping struct {
+	Source []string `json:"source"`
+	Target []string `json:"target"`
```

Comment:
The DC mapping should be single string to single string.
DC1 (of the source cluster) => DC2 (of the destination cluster) ; DC2 (of the source) => DC1 (of the target)

And that's it, no need for the complex logic.